framework/src/ics/InitialConditionTempl.C

```diff
@@ -173,11 +173,13 @@ InitialConditionTempl<T>::compute()
   // From here on out we won't be sampling at nodes anymore
   _current_node = nullptr;
 
+  const bool add_p_level = _var.dofMap().should_p_refine(_fe_type.family);
```

Comment:
After much debugging, found this was a problem I was having when it was still trying to p-refine lagrange variables. I think this is proper thing to do:

```suggestion
const bool add_p_level = _var.dofMap().should_p_refine(_var.dofMap().var_group_from_var_number(_var.number()));
```

Reply:
Oh wow, implicit conversion led to me not updating that code. I will probably delete the implicit conversion to ensure that can’t happen in the future

Reply:
sorry you had to debug that!

Reply:
opened https://github.com/libMesh/libmesh/pull/3619

Reply:
ok fixed this line